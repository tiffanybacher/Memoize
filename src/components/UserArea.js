import React, { Component } from 'react';

class UserArea extends Component {
  constructor(props) {
    super(props);
    this.state = { 
      displaySavedBtn: false
    }
  }

  moveToNextQuestion = () => {
    this.props.hideSavedMsg();
    this.props.hideAnswer();
    this.props.updateCardSelection();
  }

  saveCard = () => {
    this.props.saveCardToStorage();
    this.props.updateSavedCards();
    this.props.showSavedMsg();
    this.setState({
      displaySavedBtn: true
    });
  }

  render() {
    if (this.state.displaySavedBtn) {
      return (
          <button
            className="next-btn"
            onClick={this.moveToNextQuestion}>
            Okay! Next Question!
          </button>
      );
    } else {
      return (
        <section className="UserAnswer">
          <h3>Your Answer:</h3>
          <p>{this.props.userAnswer}</p>
          <div className="user-btn-area">
            <button 
              className="accept-btn"
              onClick={this.moveToNextQuestion}>
                I feel good about this!</button>
            <button 
              className="save-btn"
              onClick={this.saveCard}>
                I may need to come back to this.</button>
          </div>
        </section>
      );
    }
  }
}

export default UserArea;