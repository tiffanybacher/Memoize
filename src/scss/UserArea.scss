.UserAnswer {

  p {
    background-color: white;
    border-radius: $rounded-border;
    margin: 0;
    padding: 20px;
  }

  .user-btn-area {
    display: flex;
    flex-direction: column;
    margin: 15px 0 0 0;

    button {
      background-color: #999;
      border: none;
      border-radius: $rounded-border;
      color: white;
      font-family: "Montserrat";
      font-size: 1.1em;
      font-weight: 600;
      padding: 18px;
      margin: 10px 0 0 0;
    }

    .accept-btn {

      &:hover {
        background-color: $pink;
      }
    }
    
    .save-btn {

      &:hover {
        background-color: $purple;
      }
    }
  }
}

.next-btn {
  background-color: #999;
  border: none;
  border-radius: $rounded-border;
  color: white;
  font-family: "Montserrat";
  font-size: 1.1em;
  font-weight: 600;
  padding: 18px;
  margin: 10px 0 0 0;
  width: 100%;

  &:hover {
    background-color: $purple;
  }
}