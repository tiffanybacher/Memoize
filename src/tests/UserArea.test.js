import React from 'react';
import ReactDOM from 'react-dom';
import { shallow } from 'enzyme';
import UserArea from '../components/UserArea';

const mockUserAnswer = 'An object is an object.';
const mockHideAnswer = jest.fn();
const mockUpdateCardSelection = jest.fn();
const mockShowSavedMsg = jest.fn();
const mockHideSavedMsg = jest.fn();
const mockSaveCardToStorage = jest.fn();
const mockUpdateSavedCards = jest.fn();

const defaultState = {
  displayNextBtn: false
}

describe('UserArea', () => {
  let wrapper;

  beforeEach(() => {
    wrapper = shallow(
      <UserArea 
        userAnswer={mockUserAnswer}
        hideAnswer={mockHideAnswer}
        updateCardSelection={mockUpdateCardSelection}
        showSavedMsg={mockShowSavedMsg}
        hideSavedMsg={mockHideSavedMsg}
        saveCardToStorage={mockSaveCardToStorage}
        updateSavedCards={mockUpdateSavedCards} />
    );
  });

  it('should match snapshot if displayNextBtn is true', () => {
    wrapper.setState({
      displayNextBtn: true
    });

    expect(wrapper).toMatchSnapshot();
  });

  it('should match snapshot if displayNextBtn is false', () => {
    wrapper.setState({
      displayNextBtn: false
    });

    expect(wrapper).toMatchSnapshot();
  });

  it('should have a default state', () => {
    expect(wrapper.state()).toEqual(defaultState);
  });

  it.skip('should call hideSavedMsg, hideAnswer, and updateCardSelection when .next-btn or .accept-btn are clicked', () => {
    
  });

  it.skip('should call saveCard when .save-btn is clicked', () => {

  });

  it('should save card to storage and display .next-btn', () => {
    wrapper.instance().saveCard();

    expect(mockSaveCardToStorage).toBeCalled();
    expect(mockUpdateSavedCards).toBeCalled();
    expect(mockShowSavedMsg).toBeCalled();

    expect(wrapper.state('displayNextBtn')).toEqual(true);
  });
});