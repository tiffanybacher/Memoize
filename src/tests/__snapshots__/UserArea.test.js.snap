// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`UserArea should match snapshot if displayNextBtn is false 1`] = `
<section
  className="UserAnswer"
>
  <h3>
    Your Answer:
  </h3>
  <p>
    An object is an object.
  </p>
  <div
    className="user-btn-area"
  >
    <button
      className="accept-btn"
      onClick={[Function]}
    >
      I feel good about this!
    </button>
    <button
      className="save-btn"
      onClick={[Function]}
    >
      I may need to come back to this.
    </button>
  </div>
</section>
`;

exports[`UserArea should match snapshot if displayNextBtn is true 1`] = `
<button
  className="next-btn"
  onClick={[Function]}
>
  Okay! Next Question!
</button>
`;
